nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -1,21 +1,146 @@
-"""Authors: Cody Baker."""
+"""Authors: Saksham Sharda, Cody Baker."""
 from pathlib import Path
+from typing import Union, Tuple
+
 import numpy as np
-from typing import Union
+from tqdm import tqdm
+
+from ....utils.json_schema import FilePathType
 
 try:
     import cv2
 
     HAVE_OPENCV = True
 except ImportError:
     HAVE_OPENCV = False
+INSTALL_MESSAGE = "Please install opencv to use this interface! (pip install opencv-python)"
 
 PathType = Union[str, Path]
 
 
+class VideoCaptureContext:
+    """Retrieving video metadata and frames using a context manager"""
+
+    def __init__(self, file_path: FilePathType):
+        assert HAVE_OPENCV, INSTALL_MESSAGE
+        self.vc = cv2.VideoCapture(filename=file_path)
+        self.file_path = file_path
+        self._current_frame = 0
+        self._frame_count = None
+        self._movie_open_msg = "The Movie file is not open!"
+
+    def get_movie_timestamps(self):
+        """Return numpy array of the timestamps(s) for a movie file."""
+        ts2 = []
+        for no in tqdm(range(self.get_movie_frame_count()), desc="retrieving timestamps"):
+            success, frame = self.vc.read()
+            ts2.append(self.vc.get(cv2.CAP_PROP_POS_MSEC) / 1000)
+            if not success:
+                break
+        return np.array(ts2)
+
+    def get_movie_fps(self):
+        """Return the internal frames per second (fps) for a movie file"""
```

Comment:
```suggestion
        """Return the internal frames per second (fps) for a movie file."""
```
`pydocstyle D400`. I'd suggest setting up some kind of linting in your IDE to make these easier to catch.